golang/internal/helper/image/image.go

```diff
@@ -204,49 +205,80 @@ func Pull(ctx context.Context, cli client.APIClient, logger io.StringWriter, exp
 	return err
 }
 
-// CustomImagePull is a client side `smart` Pull, that only pulls if the digests are not matching
-func CustomImagePull(ctx context.Context, imageName, encodedAuth string, forcePull, preferLocal bool, displayFn PullDisplayFn) error {
+func parseDistributionRef(imageName string) (reference.Named, error) {
 	distributionRef, nameErr := reference.ParseNormalizedNamed(imageName)
 	switch {
 	case nameErr != nil:
-		return nameErr
+		return nil, nameErr
 	case reference.IsNameOnly(distributionRef):
 		distributionRef = reference.TagNameOnly(distributionRef)
 		if tagged, ok := distributionRef.(reference.Tagged); ok {
 			log.Info().Msgf("Using default tag for image %s", tagged.String())
 		}
 	}
+	return distributionRef, nil
+}
 
-	cli, cliErr := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
-	if cliErr != nil {
-		return cliErr
-	}
-
-	if !forcePull {
+func shouldUseLocalImage(ctx context.Context, cli client.APIClient,
+	distributionRef reference.Named, encodedAuth string, preferLocal bool,
+) (bool, error) {
+	if preferLocal {
```

Comment:
Feels a bit strange calling this function only to check a param, while you could check it before calling it.

Also `preferLocal` feels like it should only be considered if both local and remote images exist and then local should be used. So maybe `forceLocal` or something like that?

Reply:
As discussed it is prefer local because it attempts to use remote if local is not available.